File: src/org/symbolmath/ast/leaf/IdentifierElement.java
package org.symbolmath.ast.leaf;

import org.symbolmath.scanner.Token;

public class IdentifierElement extends LeafElement {
  private String name;

  public IdentifierElement(Token token) {
    this.name = (String) token.getValue();
  }

  @Override
  public String toString() {
    return name;
  }
}

File: src/main/java/org/symbolmath/ast/leaf/IdentifierElement.kt
package org.symbolmath.ast.leaf

class IdentifierElement(var name: String) : LeafElement() {

    override fun toString(): String {
        return name
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as IdentifierElement

        if (name != other.name) return false

        return true
    }

    override fun hashCode(): Int {
        return name.hashCode()
    }


}